applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp

```diff
@@ -348,17 +350,23 @@ void DistanceModificationProcess::RecoverDeactivationPreviousState(){
         auto it_elem = mrModelPart.ElementsBegin() + i_elem;
         it_elem->Set(ACTIVE,true);
     }
-
-    // Free the negative DOFs that were fixed
-    #pragma omp parallel for
-    for (int i_node = 0; i_node < static_cast<int>(mrModelPart.NumberOfNodes()); ++i_node){
-        auto it_node = mrModelPart.NodesBegin() + i_node;
-        if (it_node->GetValue(EMBEDDED_IS_ACTIVE) == 0){
-            // Free the nodal DOFs that were fixed
-            it_node->Free(PRESSURE);
-            it_node->Free(VELOCITY_X);
-            it_node->Free(VELOCITY_Y);
-            it_node->Free(VELOCITY_Z);
+    if ((mDoubleVariablesList.size() > 0.0) || (mComponentVariablesList.size() > 0.0)){
+        // Free the negative DOFs that were fixed
+        #pragma omp parallel for
+        for (int i_node = 0; i_node < static_cast<int>(mrModelPart.NumberOfNodes()); ++i_node){
+            auto it_node = mrModelPart.NodesBegin() + i_node;
+            if (it_node->GetValue(EMBEDDED_IS_ACTIVE) == 0){
+                for (int i_var = 0; i_var < static_cast<int>(mDoubleVariablesList.size()); i_var++){
+                    const Variable<double>& r_double_var = *mDoubleVariablesList[i_var];
```

Comment:
@jcotela, now there is no "KratosComponents" call in the nodal loop. 

Would you also move the definition of r_double_var and r_component_var outside the nodal loop?

I kept them inside since doing it outside would require to do as many nodal loops as variable in the list, but Im not quite sure which option is faster.

Reply:
I would do it like this, I think it is better to loop less times over the nodes.